server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java

```diff
@@ -859,46 +856,68 @@ public void testScalingWithEmptyCoreAndKeepAliveAndWorkerPoolProbing() {
     }
 
     private void testScalingWithEmptyCore(EsThreadPoolExecutor executor) {
```

Comment:
I think I'd prefer to keep the old tests and add these new ones alongside rather than modifying the old tests like this. It's not quite so clear that they are testing the case we really care about any more.

Reply:
👍 reverted those back to your original reproduction